Dockerfile

```diff
@@ -0,0 +1,64 @@
+FROM nvidia/cuda:11.8.0-devel-ubuntu22.04 AS dev
+
+RUN apt-get update -y \
+    && apt-get install -y python3-pip python3-venv
+
+WORKDIR /workspace
+COPY requirements.txt requirements.txt
+RUN --mount=type=cache,target=/root/.cache/pip \
+    pip install -r requirements.txt
+    
+COPY requirements-dev.txt requirements-dev.txt
+RUN --mount=type=cache,target=/root/.cache/pip \
+    pip install -r requirements-dev.txt
+
+FROM dev AS build_wheel
+
+ARG max_jobs=4
+
+COPY csrc csrc
+COPY vllm vllm
+COPY pyproject.toml pyproject.toml
+COPY README.md README.md
+COPY MANIFEST.in MANIFEST.in
+COPY setup.py setup.py
+
+RUN --mount=type=cache,target=/root/.cache/pip \
+    MAX_JOBS=$max_jobs python3 -m build
+
+FROM dev AS build
+
+COPY csrc csrc
+COPY setup.py setup.py
+COPY README.md README.md
```

Comment:
I would replace this line with CMD touch README.md

Reply:
Think it is odd that the build depends on documentation - makes sense for the release but given the amount of time it takes to build I don't think it makes sense for development builds. Think making this an optional dependency of the build process would be better